pkg/process/procutil/process_linux.go

```diff
@@ -123,6 +160,110 @@ func (p *Probe) getCmdline(pidPath string) []string {
 	return trimAndSplitBytes(cmdline)
 }
 
+// parseStatus retrieves status info from "status" file for a process in procfs
+func (p *Probe) parseStatus(pidPath string) *statusInfo {
+	path := filepath.Join(pidPath, "status")
+	var err error
+
+	sInfo := &statusInfo{
+		uids:        make([]int32, 0),
+		gids:        make([]int32, 0),
+		memInfo:     &MemoryInfoStat{},
+		ctxSwitches: &NumCtxSwitchesStat{},
+	}
+
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		return sInfo
+	}
+
+	index := 0
+	for i, r := range contents {
+		if r == '\n' {
+			p.parseStatusLine(contents[index:i], sInfo)
+			index = i + 1
+		}
+	}
+
+	return sInfo
+}
+
+// parseStatusLine takes each line in "status" file and parses info from it
+func (p *Probe) parseStatusLine(line []byte, sInfo *statusInfo) {
+	for i := range line {
+		if i+2 < len(line) && line[i] == ':' && line[i+1] == '\t' {
+			key := line[0:i]
+			value := line[i+2:]
+			p.parseStatusKV(string(key), string(value), sInfo)
+			break
+		}
+	}
+}
+
+// parseStatusKV takes tokens parsed from each line in "status" file and populates statusInfo object
+func (p *Probe) parseStatusKV(key, value string, sInfo *statusInfo) {
+	switch key {
+	case "Name":
+		sInfo.name = strings.Trim(value, " \t")
```

Comment:
Just as a curiosity, why do we need to trim by `<space>\t` instead of `\t`?

Reply:
FWIW, `value` excludes the `:\t` characters, so I think it's just a safety net to remove any whitespace. We could also use [strings.TrimSpace()](https://golang.org/pkg/strings/#TrimSpace), which first attempts to trim using pure ASCII instead of decoding to UTF-8.

Reply:
I think you misunderstood the second parameter of the `Trim` function, it's a string that contains a set of chars to remove from the beginning and end of a string. 
In status file the values are aligned using space and `\t` so we need to remove both of them:
```
VmPeak:    37908 kB
VmSize:    37676 kB
VmLck:         0 kB
```